src/cfnlint/transform.py

```diff
@@ -0,0 +1,161 @@
+  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+  OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+import os
+import six
+
+from samtranslator.parser import parser
+from samtranslator.translator.translator import Translator
+from samtranslator.public.exceptions import InvalidDocumentException
+
+import cfnlint.helpers
+
+
+class Transform(object):
+    """
+    Application Serverless Module tranform Wrappor. Based on code from AWS SAM CLI:
+    https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/commands/validate/lib/sam_template_validator.py
+    """
+
+    def __init__(self, filename, template, region):
+        """
+        Initialize Transform class
+        """
+        self._filename = filename
+        self._template = template
+        self._region = region
+
+        self._managed_policy_map = self.load_managed_policies()
+        self._sam_parser = parser.Parser()
+
+
+    def load_managed_policies(self):
+        """
+        Load the ManagedPolicies locally, based on the AWS-CLI:
+        https://github.com/awslabs/aws-sam-cli/blob/develop/samcli/lib/samlib/default_managed_policies.json
+        """
+        return cfnlint.helpers.load_resources('data/Serverless/ManagedPolicies.json')
+
+    def _replace_local_codeuri(self):
+        """
+        Replaces the CodeUri in AWS::Serverless::Function and DefinitionUri in AWS::Serverless::Api to a fake
+        S3 Uri. This is to support running the SAM Translator with valid values for these fields. If this in not done,
+        the template is invalid in the eyes of SAM Translator (the translator does not support local paths)
+        """
+
+        all_resources = self._template.get('Resources', {})
+
+        for _, resource in all_resources.items():
+
+            resource_type = resource.get('Type')
+            resource_dict = resource.get('Properties')
+
+            if resource_type == 'AWS::Serverless::Function':
+
+                Transform._update_to_s3_uri('CodeUri', resource_dict)
+
+            if resource_type == 'AWS::Serverless::Api':
+                if 'DefinitionBody' not in resource_dict:
+                    Transform._update_to_s3_uri('DefinitionUri', resource_dict)
+
+    def transform_template(self):
+        """
+        Transform the Template using the Serverless Application Model.
+        """
+        matches = []
+
+        # Boto's mechanism is used to fetch the region. Since we know which we use
+        # to lint, make sure Boto loads that one.
+        # https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/translator/arn_generator.py#L39
+        # https://github.com/awslabs/serverless-application-model/blob/master/tests/translator/test_translator.py#L120
+        old_region = os.environ.get('AWS_DEFAULT_REGION', '')
```

Comment:
Oh this kind of sucks... we have to rewrite an environment variable?  Is that what the one failed build pointed too?

Reply:
This is literally the error on the Build, or when you don't have a region specified, check the links. ATM this is an issue in SAM itself :(

Alternative could fix this in Tox

Reply:
At the moment, SAM requires a region. This is to allow for some of the wacky stuff we needed to do to hind some things away. SAM should probably just default to something sane (whatever that is). It makes sense to have the region required when in the service context when but not really when it is a library. Improvement for sure.